cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -351,12 +350,8 @@ public ApplicationDetail getLatestAppDetail(NamespaceId namespaceId,
 
     ApplicationId appId = namespaceId.app(appMeta.getSpec().getName(), appMeta.getSpec().getAppVersion());
     String ownerPrincipal = ownerAdmin.getOwnerPrincipal(appId);
-    if (enforceOwnerPrincipal) {
-      return enforceApplicationDetailAccess(appId, ApplicationDetail.fromSpec(appMeta.getSpec(), ownerPrincipal,
-                                                                              appMeta.getChange()));
-    }
-    return ApplicationDetail.fromSpec(appMeta.getSpec(), ownerPrincipal,
-                                      appMeta.getChange());
+    return enforceApplicationDetailAccess(appId, ApplicationDetail.fromSpec(appMeta.getSpec(), ownerPrincipal,
```

Comment:
Why did you change it back @chtyim ? If the user only has DELETE privilege then this would fail.